### DISCUSSION TITLE
WARNING: gf_struct should be a list of pairs '(str,int)' containing the block-name and its linear size and not a list of pairs '(str,[str or int,...])'. Converting...

### DISCUSSION THREAD

**Discussion by quantumable** [TRIQS/dft_tools#210] [General]
Hello All,

I launched the dmft calculation   [nio exemple](https://triqs.github.io/dft_tools/latest/tutorials/nio_csc.html)    with the new version and I want to know if this warning is important?

`WARNING: gf_struct should be a list of pairs '(str,int)' containing the block-name and its linear size and not a list of pairs '(str,[str or int,...])'. Converting...`

best

**Comment by hschnait** [comment]
In short no - the  form of the `gf_struct` object has been changed in 3.1.x but is converted automatically (thus the `Converting...` in your warning message).

However, I'm wondering if are you using the latest dft_tools/3.1.0 (released yesterday)?
We've updated the form of the `SK.gf_struct_solver_list` object there as well, so no warning should be visible?

**Reply by quantumable** [reply]
yes this [triqs](https://triqs.github.io/triqs/latest/install.html) I recompile it yesterday. 
And despite the warning, the calculation goes to the end and I hope this does not impact the results.
Nio tutorial [script](https://triqs.github.io/dft_tools/latest/tutorials/images_scripts/nio.py.html#nio-py) still valid?

**note:**

we should change 
 ```
# print densities
    for sig,gf in S.G_iw:
        mpi.report("Orbital %s density: %.6f"%(sig,dm[sig][0,0]))
    mpi.report('Total charge of Gloc : %.6f'%S.G_iw.total_density())

```
to 
   

```
  for sig,gf in S.G_iw:
      mpi.report("Orbital %s density: %.6f"%(sig,dm[sig][0,0].real))
  mpi.report('Total charge of Gloc : %.6f'%S.G_iw.total_density().real)
 

```

**Reply by the-hampel** [reply]
Hi, 
can you please show the output of the following python snippet: 
```
from triqs_dft_tools import version

version.show_version()
version.show_git_hash()
```
I suspect that your dft_tools version is not yet the latest one? Because I definitely tested the nio.py example and is does not give this warning anymore.

**Reply by quantumable** [reply]
```
>>> from triqs_dft_tools import version
Warning: could not identify MPI environment!
Starting serial run at: 2022-03-10 18:03:23.323162
>>> version.show_version()

You are using triqs_dft_tools version 3.1.0

>>> version.show_git_hash()

You are using triqs_dft_tools git hash 5f957c7518dffc3a122396fee5aaddc2b007c50f based on triqs git hash e24f9c8d38b4fea199a17cd240213803be81d629
```

**Reply by the-hampel** [reply]
yes indeed this version is not the latest version. You must have pulled the git repo just a few hours before we made the changes. Please pull the latest version via git pull in the src directory and try again.

**Comment by quantumable** [comment]
yes indeed there is no longer the warning when installing yesterday's version.
